### DISCUSSION TITLE
Ejabberd hostname and ssl error

### DISCUSSION THREAD

**Discussion by techteam1234** [processone/ejabberd#3696] [Questions] [answered]
Hello,

I am using Ejabberd in my server and the chat client is Gajim. I am getting 'Certificate Verification failed" error while adding account in Gajim.

We installed SSL using the below commands:

~~~
SSL : 
====
yum -y install epel-release
yum install certbot python2-certbot-apache mod_ssl
sudo certbot certonly --server https://acme-v02.api.letsencrypt.org/directory --manual --preferred-challenges dns -d *.xx.yy.net
~~~

The hostname is set as :

~~~
# cat /etc/hostname 
xx.yy.net
# cat /etc/hosts 
127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4
127.0.0.1 xx.yy.net
~~~

SSl certificate entry in ejabberd config file is as follows :+1: 
~~
hosts:
  - xx.yy.net

loglevel: info

certfiles:
  - "/opt/ejabberd/conf/server.pem"
## - "/etc/letsencrypt/live/xx.yy.net/privkey.pem"

ca_file: "/opt/ejabberd/conf/cacert.pem"
~~

Please let us know if any settings need to be changed to fix the error
<img width="361" alt="Screenshot_1 (1)" src="https://user-images.githubusercontent.com/92359530/136912229-2ff4d6ef-3a00-4e42-ad9e-cada51cf8cf3.png">
.

**Comment by techteam1234** [comment]
Hello,

Any updates?

**Comment by licaon-kter** [comment]
As the pictures show, it doesn't use the LE cert. Did you restart ejabberd? Did you mix-up the cert files?

**Comment by techteam1234** [comment]
Yes I have restarted ejabberd. whenever im making any ssl related changes in config file, ejabberd becomes down. How can I install LE certificate properly for the hostname xx.yy.net?
I used this command before:
certbot certonly --server https://acme-v02.api.letsencrypt.org/directory --manual --preferred-challenges dns -d *.xx.yy.net

**Reply by techteam1234** [reply]
Hello,
 
Shall we get any update on this

**Comment by licaon-kter** [comment]
What does `"/opt/ejabberd/conf/server.pem"` contain?

**Reply by techteam1234** [reply]
It contains the private key and certificate.

**Comment by licaon-kter** [comment]
An the other 3 files?

**Comment by techteam1234** [comment]
/etc/letsencrypt/live/xx.yy.net/fullchain.pem >> It has the certs too
/etc/letsencrypt/live/xx.yy.net/privkey.pem >> Has the private key.

**Comment by licaon-kter** [comment]
The same certs in 3 files? And the CAs in another?#

Can you limit to only 2 files? Make sure `ejabberd` the user can access the paths, all the subdirs, and the files.

What does ejabberd.log say on start? Paste it

**Reply by techteam1234** [reply]
Hello,

Thanks for your reply,

SSL configuration in ejabberd.yml,

hosts:

XXXX.net
loglevel: info

certfiles:

"/opt/ejabberd/conf/server.pem"
"/etc/letsencrypt/live/XXX/fullchain.pem"
"/etc/letsencrypt/live/XXX/privkey.pem"
ca_file: "/opt/ejabberd/conf/cacert.pem"

After saving, when restart ejabberd, Getting the below error.

service ejabberd status
Getting ejabberd status...
Failed RPC connection to the node ejabberd@localhost: nodedown

Commands to start an ejabberd node:
start Start an ejabberd node in server mode
debug Attach an interactive Erlang shell to a running ejabberd node
iexdebug Attach an interactive Elixir shell to a running ejabberd node
live Start an ejabberd node in live (interactive) mode
iexlive Start an ejabberd node in live (interactive) mode, within an Elixir shell
foreground Start an ejabberd node in server mode (attached)

Optional parameters when starting an ejabberd node:
--config-dir dir Config ejabberd: /opt/ejabberd/conf
--config file Config ejabberd: /opt/ejabberd/conf/ejabberd.yml
--ctl-config file Config ejabberdctl: /opt/ejabberd/conf/ejabberdctl.cfg
--logs dir Directory for logs: /opt/ejabberd/logs
--spool dir Database spool dir: /opt/ejabberd/database/ejabberd@localhost
--node nodename ejabberd node name: ejabberd@localhost

Please check and help us solve the issue as soon as possible. Is there any way to connect live chat with here to discuss and solve the problem

**Comment by licaon-kter** [comment, marked as answer]
ejabberd.log please
